libs/langchain/langchain/chat_models/azure_openai.py

```diff
@@ -137,7 +146,19 @@ def _create_chat_result(self, response: Mapping[str, Any]) -> ChatResult:
         for res in response["choices"]:
             if res.get("finish_reason", None) == "content_filter":
                 raise ValueError(
-                    "Azure has not provided the response due to a content"
-                    " filter being triggered"
+                    "Azure has not provided the response due to a content filter "
+                    "being triggered"
                 )
-        return super()._create_chat_result(response)
+        chat_result = super()._create_chat_result(response)
+
+        if "model" in response:
+            model = response["model"]
+            if self.model_version:
+                model = f"{model}-{self.model_version}"
+
+            if chat_result.llm_output is not None and isinstance(
+                chat_result.llm_output, dict
+            ):
+                chat_result.llm_output["model_name"] = model
```

Comment:
i think having model and model_name is a bit confusing. maybe should just include "model_version" directly?

Reply:
oh but this is needed for cost counting

Reply:
yes, by default AzureChatOpenAI was returning a hard-coded `gpt-35-turbo` no matter what was the actual model used. Also AzureChatOpenAI use _deployment_name_ which doesn't have to relate to model name at all. Thus I thought it would be best capture and use the model returned from the API.